Competition/src/main/cpp/Drivetrain.cpp

```diff
@@ -7,15 +7,231 @@
+  m_rightPIDController.SetI(DriveConstants::kI);
+  m_rightPIDController.SetD(DriveConstants::kD);
+  m_rightPIDController.SetIZone(DriveConstants::kIz);
+  m_rightPIDController.SetFF(DriveConstants::kFF);
+  m_rightPIDController.SetOutputRange(DriveConstants::kMinOutput, DriveConstants::kMaxOutput);
+
+  m_rightPIDController.SetSmartMotionMaxVelocity(DriveConstants::kMaxVel);
+  m_rightPIDController.SetSmartMotionMinOutputVelocity(DriveConstants::kMinVel);
+  m_rightPIDController.SetSmartMotionMaxAccel(DriveConstants::kMaxAccel);
+  m_rightPIDController.SetSmartMotionAllowedClosedLoopError(DriveConstants::kAllError);
+  
+  m_leftPIDController.SetSmartMotionMaxVelocity(DriveConstants::kMaxVel);
+  m_leftPIDController.SetSmartMotionMinOutputVelocity(DriveConstants::kMinVel);
+  m_leftPIDController.SetSmartMotionMaxAccel(DriveConstants::kMaxAccel);
+  m_leftPIDController.SetSmartMotionAllowedClosedLoopError(DriveConstants::kAllError);
+
+  m_leftDriveLead.SetInverted(false);
+  m_rightDriveLead.SetInverted(true);
+}
+
+void Drivetrain::ResetEncoders() {
+  m_leftEncoder.SetPosition(0);
+  m_rightEncoder.SetPosition(0);
+}
+
+void Drivetrain::ResetOdometry(frc::Pose2d pose) {
+  ResetEncoders();
+  m_odometry.ResetPosition(pose, frc::Rotation2d(units::degree_t(GetHeading())));
+}
+void Drivetrain::ResetIMU() {
+  imu.Reset();
+}
+
+// rev::CANEncoder& GetLeftEncoder() {
+//     return m_leftEncoder;
+// }
+
+// rev::CANEncoder& GetRightEncoder() {
+//     return m_rightEncoder;
+// }
+
+double Drivetrain::GetEncAvgDistance() {
+  return ((m_leftEncoder.GetPosition() * RamseteConstants::kPositionConversionFactor) + (m_rightEncoder.GetPosition() * RamseteConstants::kPositionConversionFactor)) / 2.0;
+}
+
+units::meter_t Drivetrain::GetLeftDistance() {
+  return m_leftEncoder.GetPosition() * RamseteConstants::kPositionConversionFactor * 1_m;
+}
+
+units::meter_t Drivetrain::GetRightDistance() {
+  return m_rightEncoder.GetPosition() * RamseteConstants::kPositionConversionFactor * 1_m;
+}
+
+double Drivetrain::GetHeading() {
+  return std::remainder(imu.GetAngle(), 360) * (RamseteConstants::kGyroReversed ? -1.0 : 1.0);
+}
+
+double Drivetrain::GetTurnRate() {
+  return imu.GetRate() * (RamseteConstants::kGyroReversed ? -1.0 : 1.0);
+}
+
+frc::Pose2d Drivetrain::GetPose() { 
+  return m_odometry.GetPose(); 
+}
+
+frc::DifferentialDriveWheelSpeeds Drivetrain::GetWheelSpeeds() {
+  return { units::meters_per_second_t(m_leftEncoder.GetVelocity() * RamseteConstants::kVelocityConversionFactor),
+            units::meters_per_second_t(m_rightEncoder.GetVelocity() * RamseteConstants::kVelocityConversionFactor) };
+}
+
+void Drivetrain::TankDriveVolts(units::volt_t leftVolts, units::volt_t rightVolts) {
+  m_leftDriveLead.SetVoltage(leftVolts);
+  m_rightDriveLead.SetVoltage(rightVolts);
+  // m_drive.Feed();
+}
+
+void Drivetrain::ArcadeDrive(double leftInput, double rightInput) {
+  // drive.ArcadeDrive(leftInput, -rightInput * .5, true);
```

Comment:
We're not using it because of rocket league controls so we can delete it for now. If we ever use arcade drive, we will have to also create DifferentialDrive object so we can do that if needed in the future.